File: src/main/java/net/matees/settings/IntegerSetting.java
package net.matees.settings;

import java.util.List;

public abstract class IntegerSetting extends Setting<Integer>{
    @Override
    public void setIntValue(int value) { setSetting(value); System.out.println(value); }

    @Override
    public void setBooleanValue(boolean value) { }

    @Override
    public void setStringValue(String value) { }

    @Override
    public void setListValue(List value) { }
}

File: src/main/java/net/matees/settings/IntegerSetting.kt
package net.matees.settings

import org.bukkit.entity.Player

abstract class IntegerSetting : Setting<Int?>() {
    open val min = 1

    override fun setIntValue(value: Int) {
        if(value < min) return
        setting = value
    }

    override fun setBooleanValue(value: Boolean) {
    }

    override fun setStringValue(value: String?) {
    }

    override fun setPlayersValue(value: List<Player?>?) {
    }

}
